txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +69,21 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            transctionToUidMappings.putIfAbsent(transaction, new Uid());
```

Comment:
Why don't you use the hash code of the transaction since that does not change no matter how the transaction gets wrapped (see section 3.3.4 Transaction Equality and Hash Code of the JTA spec). That way you can avoid having to maintain these two maps (and any race conditions you were worried about)?